pex/tools/commands/venv.py

```diff
@@ -61,12 +63,206 @@ def _copytree(
                         link = False
                 shutil.copy(src_entry, dst_entry)
             except OSError as e:
-                if e.errno == errno.EEXIST:
-                    pex_warnings.warn(
-                        "Failed to overwrite {} with {}: {}".format(dst_entry, src_entry, e)
+                if e.errno != errno.EEXIST:
+                    raise e
+
+
+class CollisionError(Exception):
+    """Indicates multiple distributions provided the same file when merging a PEX into a venv."""
+
+
+def populate_venv_with_pex(
```

Comment:
This was mainly extracted from the Venv Command so that pex_bootstrapper could use it. The only real change was to add `record_provenance` which I used to track down a collision in building a venv from Toolchain requirements. In that case two different distributions were providing the same script - `jsondiff` (one from https://pypi.org/project/jsondiff/ and one from https://pypi.org/project/jsonpatch/). The `bin/` scripts are namespaced even more loosely than the global project package namespace (basically not at all) so this seems likely to keep happening. That drove me to default collisions being ok for the `--venv` mode and just emitting warnings.

Reply:
> That drove me to default collisions being ok for the --venv mode and just emitting warnings.

It looks like the default for the option is `False`?

Reply:
Nope, `collisions_ok=True` is the default.